### PR TITLE
IGNITE-22746 DB API Driver 3: Implement documentation generation

### PR DESCRIPTION
**Pull request by isapego**
https://issues.apache.org/jira/browse/IGNITE-22746

- Added dependencies for docs;
- Implemented generation of the documentation;
- Added instruction to generate docs to README.md.

### CODE REVIEW

File: modules/platforms/python/README.md [hunk 1]
```diff
@@ -46,3 +46,20 @@ Running tests themselves:
 ```bash
 $ pytest
 ```
+
+## Documentation
+
+Do not forget to install documentation requirements:
```
Comment (ptupitsyn) [inline modules/platforms/python/README.md:52]:
```suggestion
Install documentation requirements: 
```

File: modules/platforms/python/README.md [hunk 2]
```diff
@@ -46,3 +46,20 @@ Running tests themselves:
...
 ```
+
+## Documentation
+
+Do not forget to install documentation requirements: 
+```bash
+$ pip install -r requirements/docs.txt
+```
+
+After this, you can generate documentation using following commands:
```
Comment (ptupitsyn) [inline modules/platforms/python/README.md:57]:
```suggestion
Generate documentation:
```

I think shorter is better here.

File: modules/platforms/python/docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+sys.path.insert(0, os.path.abspath('../'))
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'pyignite3'
+copyright = '2024, The Apache Software Foundation'
+author = 'The Apache Software Foundation'
+release = '3.1'
```
Comment (ptupitsyn) [inline modules/platforms/python/docs/conf.py:32]: Is this the Ignite version? Should we update the release procedure to bump this number?

**Review by ptupitsyn — APPROVED** [review]

File: modules/platforms/python/README.md (hunk 1 above)
Comment (isapego) [reply modules/platforms/python/README.md:52]: Fixed.

File: modules/platforms/python/README.md (hunk 2 above)
Comment (isapego) [reply modules/platforms/python/README.md:57]: Agree, Simpler is better.

File: modules/platforms/python/docs/conf.py (hunk 3 above)
Comment (isapego) [reply modules/platforms/python/docs/conf.py:32]: To be handled here: https://issues.apache.org/jira/browse/IGNITE-23443

Comment (isapego) [reply modules/platforms/python/docs/conf.py:32]: Replaced with `pyignite3.__version__`

**Pull request merged**